Service order notes get all (#376)
MewsSystems/gitbook-connector-api: operations/serviceorders.md

# Service orders

> 

Reviewer: I think it would be helpful to insert a note here. Please read the wording and confirm that my understanding is correct.

> **Service Orders:** A Service Order is an Order made against a Service. There may be different types of Service Order, but the most common is a Reservation, which is an Order made against a Bookable Service. Note that a Reservation ID can be used as a Service Order ID in Service Order operations such as [Get all service order notes](#get-all-service-order-notes).

Author: Left blank on purpose, you don't have those new endpoint available yet

## Get all service order notes

> Returns all notes associated with the service order. This operation uses [Pagination](../guidelines/pagination.md).

Reviewer: ```suggestion
Returns all notes associated with the service order (e.g. reservation). This operation uses [Pagination](../guidelines/pagination.md).
```

### Request

`[PlatformAddress]/api/connector/v1/serviceOrderNotes/getAll`

```javascript
{
    "ClientToken": "E0D439EE522F44368DC78E1BFB03710C-D24FB11DBE31D4621C4817E028D9E1D",
    "AccessToken": "C66EF7B239D24632943D115EDE9CB810-EA00F8FD8294692C940F6B5A8F9453D",
    "Client": "Sample Client 1.0.0",
    "ServiceOrderIds": [
        "1dc0c6bf-2ce6-4a9f-af97-af5c01676720"
    ],
    "ServiceOrderNoteIds": [
        "a06a225b-00f7-48c8-a463-af5c016768e9"
    ],
    "Limitation": { "Count": 1 }
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `ClientToken` | string | required | Token identifying the client application. |
| `AccessToken` | string | required | Access token of the client application. |
| `Client` | string | required | Name and version of the client application. |
| `ServiceOrderIds` | array of string | required, max 1000 items | Unique identifiers of [Service order](#service-order). |

> | `ServiceOrderNoteIds` | array of string | optional, max 1000 items | Unique identifiers of [Service order notes](#service-order-note). Use this property if you want to fetch specific service order note. |

Reviewer 2: ```suggestion
| `ServiceOrderNoteIds` | array of string | optional, max 1000 items | Unique identifiers of [Service order notes](#service-order-note). Use this property if you want to fetch specific service order notes. |
```

| `Limitation` | [Limitation](../guidelines/pagination.md#limitation) | required | Limitation on the quantity of Service order notes returned. |

### Response

```javascript
{
    "ServiceOrderNotes": [
        {
            "Id": "a06a225b-00f7-48c8-a463-af5c016768e9",
            "ServiceOrderId": "1dc0c6bf-2ce6-4a9f-af97-af5c01676720",

>             "Text": "This should be an order note, right?",

Reviewer 2: ```suggestion
            "Text": "Shaken, not stirred.",
```
Kinda tricky what to put here as an example... what about this?

Author: I just put in a text from my response within my localhost lmao 😆

            "Type": "General",
            "CreatedUtc": "2022-11-29T21:48:32Z"
        }
    ],
    "Cursor": "a06a225b-00f7-48c8-a463-af5c016768e9"
}
```

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |

> | `ServiceOrderNotes` | array of [Service order note](#service-order-note) | required | The collection of Service order notes. |

Reviewer 2: ```suggestion
| `ServiceOrderNotes` | array of [Service order note](#service-order-note) | required | The collection of service order notes. |
```

| `Cursor` | string | required | Unique identifier of the last and hence oldest service order note returned. This can be used in [Limitation](../guidelines/pagination.md#limitation) in a subsequent request to fetch the next batch of older service order notes. |

#### Service order
| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |

> | `Id` | string | required | Unique identifier of the service order. |

Author: Is this enough for now? This will probably get extended once You guys start developing `serviceOrders/getAll`

Reviewer 2: I think it's fine, maybe if we have an issue already for it, put a note there.

#### Service order note

| Property | Type | Contract | Description |
| :-- | :-- | :-- | :-- |
| `Id` | string | required | Unique identifier of the service order note. |

> | `ServiceOrderId` | string | required | Unique identifier of a [Service order](#service-order). |

Reviewer: ```suggestion
| `ServiceOrderId` | string | required | Unique identifier of the [Service order](#service-order) to which the Service Order Note belongs. |
```

| `Text` | string | required | Content of the service order note. |

> | `Type` | string | required | A discriminator specifying the [type of service order note](#type), e.g. general or channel manager. |

Reviewer: Small suggestion, not sure, maybe just me being fussy - would it be better to use #note-type or #service-order-note-type rather than just #type? and also to make the link reflect the name exactly, e.g. [note type](#note-type)?

> | `Type` | string | optional | A discriminator specifying the [type of service order note](#type), e.g. general or channel manager. |

Author: This should probably be required, right?

Reviewer 2: Yes, even in DB it's not null which means 0 = General. Am I right?

Author: ```suggestion
| `Type` | string | required | A discriminator specifying the [type of service order note](#type), e.g. general or channel manager. |
```

| `CreatedUtc` | string | required | Creation date and time of the block in UTC timezone in ISO 8601 format. |

#### Type

> * General

Reviewer: ```suggestion
* `General`
```

> #### Type

Reviewer 2: It's up for a discussion, I would say yes.

> #### Type

Reviewer 2: @vitezp do you have any idea?

> #### Type

Author: @VBuben it's such a trivial thing to change that I don't think it matters as much

> #### Type

Reviewer 2: It's a trivial thing but in theory breaking change 🤷 but still it needs others to review

> #### Type

Reviewer: Do you mean if **Type** will be extended? There is a convention to represent it like this (with ellipses) if there may be other values added:
* `General`
* `ChannelManager`
* ...

> * ChannelManager

Reviewer: ```suggestion
* `ChannelManager`
```
